tmol/score/lk_ball/potentials/lk_ball.hh

```diff
@@ -478,8 +492,870 @@ struct lk_ball_score {
+      block_pair_dat.global_params);
+
+  auto accum_derivs1 = ([&] TMOL_DEVICE_FUNC(
+                            LKBallSingleResData<Real> const &block_dat,
+                            int atom_ind,
+                            Real3 dV,
+                            lk_ball_score_type st) {
+    for (int j = 0; j < 3; ++j) {
+      if (dV[j] != 0) {
+        accumulate<Dev, Real>::add(
+            dV_d_pose_coords[block_pair_dat.pose_ind]
+                            [block_dat.block_coord_offset + atom_ind][j],
+            dTdV_local[st] * dV[j]);
+      }
+    }
+  });
+
+  auto accum_derivs4 = ([&] TMOL_DEVICE_FUNC(
+                            LKBallSingleResData<Real> const &block_dat,
+                            int atom_ind,
+                            lk_ball_dV_dReal3<Real> const &dV) {
+    accum_derivs1(block_dat, atom_ind, dV.d_lkball_iso, w_lk_ball_iso);
+    accum_derivs1(block_dat, atom_ind, dV.d_lkball, w_lk_ball);
+    accum_derivs1(block_dat, atom_ind, dV.d_lkbridge, w_lk_bridge);
+    accum_derivs1(block_dat, atom_ind, dV.d_lkbridge_uncpl, w_lk_bridge_uncpl);
+  });
+
+  auto water_accum_derivs1 = ([&] TMOL_DEVICE_FUNC(
+                                  LKBallSingleResData<Real> const &block_dat,
+                                  int atom_ind,
+                                  int water_ind,
+                                  WatersMat dV,
+                                  lk_ball_score_type st) {
+    for (int j = 0; j < 3; ++j) {
+      if (dV(water_ind, j) != 0) {
+        accumulate<Dev, Real>::add(
+            dV_d_water_coords[block_pair_dat.pose_ind]
+                             [block_dat.block_coord_offset + atom_ind]
+                             [water_ind][j],
+            dTdV_local[st] * dV(water_ind, j));
+      }
+    }
+  });
+
+  auto water_accum_derivs4 = ([&] TMOL_DEVICE_FUNC(
+                                  LKBallSingleResData<Real> const &block_dat,
+                                  int atom_ind,
+                                  int water_ind,
+                                  lk_ball_dV_dWater<Real, MAX_N_WATER> const
+                                      &dV) {
+    water_accum_derivs1(
+        block_dat, atom_ind, water_ind, dV.d_lkball_iso, w_lk_ball_iso);
+    water_accum_derivs1(block_dat, atom_ind, water_ind, dV.d_lkball, w_lk_ball);
+    water_accum_derivs1(
+        block_dat, atom_ind, water_ind, dV.d_lkbridge, w_lk_bridge);
+    water_accum_derivs1(
+        block_dat, atom_ind, water_ind, dV.d_lkbridge_uncpl, w_lk_bridge_uncpl);
+  });
+
+  accum_derivs4(polar_block_dat, polar_start + polar_atom_tile_ind, dV.dI);
+  accum_derivs4(
+      occluder_block_dat, occluder_start + occluder_atom_tile_ind, dV.dJ);
+  for (int i = 0; i < MAX_N_WATER; ++i) {
+    water_accum_derivs4(
+        polar_block_dat, polar_start + polar_atom_tile_ind, i, dV.dWI);
+    water_accum_derivs4(
+        occluder_block_dat, occluder_start + occluder_atom_tile_ind, i, dV.dWJ);
+  }
+}
+
```

Comment:
maybe give better names that *4 / *1 ?

Reply:
Hmm. Gonna back-burner this name change suggestion and wrap up the PR but I'll make sure I get it on the next revision.